python/rikai/mixin.py

```diff
@@ -90,6 +90,51 @@ def __matmul__(self, style: Union[dict, "rikai.viz.Style"]) -> Drawable:
         return style(self)
 
 
+class ComposableDrawable:
+    """Mixin for a drawable that can be composed with others"""
+
+    @abstractmethod
+    def xy_min(self):
+        """return a tuple of the left top axis (xmin, ymin)"""
+
+    @abstractmethod
+    def aligned(
+        self, base_drawable: ComposableDrawable, algorithm: str = "left-top"
+    ):
+        """return a new instance that is aligned with the base drawable,
+         currently it has only `left-top` align, will
+        add more if we need them"""
+
+
+class ComposedDrawable(Drawable):
+    """A drawable that includes multi sub composable drawables,
+    will align them in rendering"""
+
+    def align(self):
+        if len(self.inner_drawables) <= 1:
+            return
+        self.inner_drawables = self.inner_drawables[:1] + [
```

Comment:
please add unit tests when `len(self.inner_drawables) > 2`

Reply:
In our current usage, the `len` would not `>2`, because it's only a box2d and a label, if we should add some use cases that have a larger drawable count, we'd better find out a concrete scenario.